packages/e2e-tests/specs/editor/blocks/list.test.js

```diff
@@ -507,4 +507,36 @@ describe( 'List', () => {
 
 		expect( await getEditedPostContent() ).toMatchSnapshot();
 	} );
+
+	it( 'should not change the contents when you change the list type to Ordered', async () => {
+		await clickBlockAppender();
+		await page.keyboard.type( '* 1' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '2' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '3' );
+		await clickBlockToolbarButton( 'Ordered' );
+
+		const content = await page.$eval(
+			'.wp-block-list',
+			( el ) => el.innerHTML
+		);
+		expect( content ).toMatchSnapshot();
```

Comment:
Could you use `await getEditedPostContent()` here?

Reply:
This is what we use everywhere else to check the edited HTML content 😄

Reply:
@ellatrix 
This change could not be tested with `getEditedPostContent`. Testing with `getEditedPostContent` passes on the current trunk.

This is because the state of the `editor` has been changed, but the DOM is not rendered correctly, which causes this problem to persist.

Reply:
It could if you type another character (make another change)? Anyway, this is fine too :)